src/Collections/AbstractValidCollection.php

```diff
@@ -2,84 +2,38 @@
 
 namespace Wickedreports\ApiCore\Collections;
 
+use ArrayIterator;
 use Wickedreports\ApiCore\Exception\ValidationException;
 
-abstract class AbstractValidCollection implements
-    \Iterator,
-    \Countable,
-    \ArrayAccess
+abstract class AbstractValidCollection extends ArrayIterator
 {
-    protected array $collection = [];
-    private int $position = 0;
-
     /**
      * @param iterable $items
      * @throws ValidationException
      */
     public function __construct(iterable $items = [])
     {
-        $this->setCollection($items);
-    }
-
-    public function current()
-    {
-        return $this->offsetGet($this->position);
-    }
-
-    public function last()
-    {
-        return $this->offsetGet($this->count() - 1);
-    }
-
-    public function next(): void
-    {
-        $this->position++;
-    }
-
-    public function key(): int
-    {
-        return $this->position;
-    }
-
-    public function valid(): bool
-    {
-        return $this->offsetExists($this->position);
```

Comment:
There is a problem. When we call unsetOffest we delete collection key and valid method return false after deleted item

Reply:
ArrayIterator has the correct logic for that, so I chose to use it.